### PR TITLE
Add Banner Bear create image action

### PR DESCRIPTION
**Pull request by vunguyenhung**
This PR adds BannerBear create an image action. Related to this issue https://github.com/PipedreamHQ/pipedream/issues/753

BannerBear API Doc can be found here: https://developers.bannerbear.com/#introduction

BannerBear Create Image API Doc can be found here: https://developers.bannerbear.com/#post-v2-images

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by dylburger** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

File: components/bannerbear/actions/create-image/create-image.mjs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        "template",
+      ],
+    },
+    modifications: {
+      type: 'object',
+      description: 'A list of modifications you want to make. [See the docs](https://developers.bannerbear.com/#post-v2-images)'
+    }
+  },
+  async run({ $ }) {
+    const response = await axios($, {
```
Comment (vellames-turing) [inline components/bannerbear/actions/create-image/create-image.mjs:25]: Could you please move your call to the `app`? Normally we use the `app` as a repository for all API calls. Take a look at `blogger` components for a reference

**Review by vellames-turing — CHANGES_REQUESTED** [review]
Hey! Thank you so much for your contribution. The main code looks good. I would like to suggest some things:
Could you please move your call at `create-image.mjs` to the `app`? Normally we use the `app` as a repository for all API calls. Take a look at `blogger` components for a reference. I would like to suggest you do some changes to `app` as well using `blogger.app.mjs` as a reference. Please let me know if you have some questions

**Comment by vunguyenhung** [comment]
Hi @vellames-turing, I have addressed your comment on the latest commit. Kindly review again when you have time. Thanks 🙏

**Review by vellames-turing — DISMISSED** [review]
Looks great. Thanks!

**Comment by DilanAthukorala** [comment]
@vunguyenhung Observed couple of issues while testing this and also the UX doesn't support sending array of objects in a single request 
example:
"modifications": [
    {
      "name": "pretitle",
      "text": "You can change this text",
      "color": null,
      "background": null
    },
    {
      "name": "title",
      "text": "You can change this text",
      "color": null,
      "background": null
    },
    {
      "name": "subtitle",
      "text": "You can change this text",
      "color": null,
      "background": null
    }
  ]



Templates are not listed when there's already created templates in the project

image



Getting this error when sending update parameters

image

**Comment by vunguyenhung** [comment]
Hi @DilanAthukorala,
I have fixed the template async option
Screen Shot 2022-04-29 at 08 29 11


And also I have changed the `modifications` option type from `object` to `string`. I've tested and the image is created successfully
Screen Shot 2022-04-29 at 08 28 54


Additionally I have added the user-friendly user message in case that the modifications field is incorrect
Screen Shot 2022-04-29 at 08 32 58

@vellames-turing Kindly review the code again when you have time 🙏

File: components/bannerbear/actions/create-image/create-image.mjs [hunk 2]
```diff
@@ -4,7 +4,7 @@ export default {
   key: "bannerbear-create-image",
   name: "Create an Image",
   description: "Create an image using template and modifications. [See the docs](https://developers.bannerbear.com/#post-v2-images)",
-  version: "0.0.1",
+  version: "0.0.6",
```
Comment (vellames-turing) [inline components/bannerbear/actions/create-image/create-image.mjs:7]: The version is wrong :)

**Review by vellames-turing — CHANGES_REQUESTED** [review]

**Review by vellames-turing — DISMISSED** [review]

**Comment by dannyroosevelt** [comment]
@vellames-turing I think this PR is out of date with master -- can you see if that fixes the errors? Then we can ship it.

**Comment by vunguyenhung** [comment]
Sorry I have pushed incorrect commits into the PR, I will create another PR for it to be cleaner and up to date with pipedream upstream 🙏

**Pull request closed without merging**